Add metabase_field semantic_type migration

lbrdnk opened the pull request:
Closes #18754

### Description

Sets semantic_type to null for metabase_field if value was type/Number, to avoid log warnings.

### How to verify

Run newly added test.

### Checklist

- [X] Tests have been added/updated to cover changes in this PR

lbrdnk committed:
Update migration and add test

Sets semantic_type to null for metabase_field if value was type/Number,
to avoid log warnings.

metamben commented on test/metabase/db/schema_migrations_test.clj:
```diff
@@ -1156,3 +1156,29 @@
         (testing "Rollback to the previous version should restore the column column, and set the default color value"
           (is (= "#31698A" (:color (t2/select-one :model/Collection :id collection-id)))))))))
+
+(deftest set-number-semantic-type-null-test
```
I feel like this is an overkill.
- The thing it tests is quite trivial and it's not huge problem if it doesn't work.
- The thing that  is potentially dangerous (e.g., that it doesn't update a bunch of unrelated things) is not being tested.
- This is not a test we need to prevent regressions or specify some behavior.
- Migration tests are particularly slow.
- Manual tests (for h2, postgres, and mysql if you want to be very through) would be enough and possibly better.

lbrdnk replied:
Thank you for the thoughtful feedback. I will mind that while writing migrations in future. As of now, test has been removed, as advised, and modified functionality tested manually for postgres, mysql and h2 as follows:

1. Start app.
2. Migrate down using `(dev/migrate! :down 47)`.
3. Set semantic_type for all rows having it null to 'type/Number'
4. Migrate up using `(dev/migrate! :up 48)`
5. Check whether 'type/Number' fields are reset to null.

metamben reviewed (commented):
I suggest removing the test. Feel free to get a second opinion if in doubt.

lbrdnk committed:
Remove test for trivial migration as suggested

As migration tests are costly and change is trvial, it is better
to test small migration as one in question manually. Hence removing
the test.

lbrdnk committed:
Merge remote-tracking branch 'origin/master' into 18754-semantic-type-migration

gonchik commented:
Hi! does that PR in the 0.47.4 release ? :) 
or it's gonna be in 0.47.5?
